packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -103,7 +104,7 @@ export function makeRumPublicApi<C extends RumInitConfiguration>(startRumImpl: S
       return
     }
 
-    if (!isValidInitConfiguration(initConfiguration)) {
+    if (willSyntheticsInjectRum() || !isValidInitConfiguration(initConfiguration)) {
```

Comment:
By adding the check here, we still check `canHandleSession`. So we might have init cookie error message even if the SDK won't be started.

Reply:
Good point, I'll move it